File: Common/Action/src/main/java/com/greentree/action/observer/pair/PairAction.java
package com.greentree.action.observer.pair;

import java.util.function.BiConsumer;

import com.greentree.action.MultiAction;


public class PairAction<T1, T2> extends MultiAction<BiConsumer<? super T1, ? super T2>> implements IPairAction<T1, T2> {
	private static final long serialVersionUID = 1L;

	@Override
	public void event(T1 t1, T2 t2) {
		for(var l : listeners)
			l.accept(t1, t2);
	}

}

File: Commons/Action/src/main/java/com/greentree/commons/action/observer/pair/PairAction.kt
package com.greentree.commons.action.observer.pair

import com.greentree.commons.action.MultiAction

class PairAction<T1, T2> : MultiAction<(T1, T2) -> Unit>(), IPairAction<T1, T2> {

	override fun event(t1: T1, t2: T2) {
		for(l in listeners)
			l(t1, t2)
	}

	companion object {

		private const val serialVersionUID = 1L
	}
}
